### ISSUE TITLE
Vehicle Speed always 0 when far away

### ISSUE THREAD

**Issue by arfla** [porisius/FicsitRemoteMonitoring#41]
Another issue that we noticed...

The vehicle speed changes to 0 after a certain distance is between the player and the vehicle. This does not happen with the train, any idea what could possibly cause that?

Thanks in advance,
arfla

**Comment by arfla** [comment]
Ah, that would indeed make sense why it was not showing up anymore. 

Haha I can imagine, I did encounter some issues with getFactory when loading into our main big world, I think it couldn't handle it performance-wise.

**Comment by porisius** [comment]
Interesting, I had fought performance issues with getFactory for a long time. Ended up using a save from BLAndrew's ridiculous world. That was a 6-7MB JSON text file. Subsystems, at least in this game, are so much better, but they didn't exist when I first made them or at least weren't exposed.

**Comment by porisius** [comment]
Doubt it will fix it, there's something either that I need to, or CSS does, to use the FGVehicleSubsystem better. I tried something, but my knowledge of C++ is pretty much non-existent.

**Comment by arfla** [comment]
I just had a quick try now with the newest update, unfortunately, that did not fix it, it actually broke the speed completely. I have attached a screenshot below, if you look at the engineRPM you can see the vehicle is moving but no speed is being output. I have tried this with autopiloted and manually driven cars. Thanks in advance for your time and effort. 

![image](https://user-images.githubusercontent.com/46490532/219853622-88b20d1e-ba14-4718-b5d5-58fb6adfea8a.png)

**Comment by porisius** [comment]
You sure about that? The Throttle is 0, and the gear is in what I would think is either Park or Neutral, so I would expect speed to be 0.0.

**Comment by arfla** [comment]
Yep pretty sure as we have got the data hooked up live on our dashboard so it updates every 100ms and none of the vehicles update anymore. Even my autopiloted vehicle says 0 speed. 

![image](https://user-images.githubusercontent.com/46490532/219882762-46c17f30-bca1-4772-8b25-e8e6967a4bbf.png)

**Comment by porisius** [comment]
Okay, that just about proves it. I am going to see about asking the CSS Devs about adding in some kind of header. I also have been trying to clean up the number of duplicate circuits anyways.

**Comment by arfla** [comment]
Speed is showing again since the latest update, the issue that the speed is disappearing after some distance with the player remains.

**Comment by porisius** [comment]
Awaiting Update 8 rework of FGVehicleSubsystem
